api_app/db/repositories/workspaces.py

```diff
@@ -70,6 +67,24 @@ def create_workspace_item(self, workspace_input: WorkspaceInCreate, auth_info: d
 
         return workspace
 
+    def get_address_space_based_on_size(self, workspace_properties: dict):
+
+        # Default the address space to 'small'. This was the pre-existing default
+        address_space_size = workspace_properties.get("address_space_size", "small")
+
+        #773 allow custom sized networks to be requested
+        if (address_space_size == "custom"):
+            # if address_space isn't provided in the input, generate a new one.
+            # TODO: #772 check that the provided address_space is available in the network.
+            return workspace_properties.get("address_space") or self.get_new_address_space()
+
+        # We allow the users some predefined TShirt sizes
+        predefined_address_spaces = { "small": 24, "medium": 22, "large": 16 }
```

Comment:
FYI (no change required) but Python 3.10 has `match select` which would have make this a little easier.

I wonder if it makes more sense to push the 'predefined_address_spaces` variable up into a constant, defined outside of this method. It's not a big deal, but at present it's being defined per call.

TBH, this might be better as an `if/elif/else` statement given that there are only two options (medium and large) that result in anything other than the `else`.

Reply:
My next ticket is validating the custom address range, so some of this will change. Personally I don't like if/elif and would prefer to use the dictionary as it's extensible and reads better